src/components/MarketDetail/index.js

```diff
@@ -104,7 +104,25 @@ class MarketDetail extends Component {
       marketFetchError: undefined,
     }
   }
+
   componentWillMount() {
+    if (!this.props.params.view) {
+      this.props.changeUrl(`/markets/${this.props.params.id}/${DEFAULT_VIEW}`)
```

Comment:
does the `changeUrl` stops the execution?

Reply:
In that case, would it make sense to introduce a `return`? or the lines 114-118 should be executed independently we go to another route?

Reply:
@apanizo this is not an another route, this is default expandable tab on the same page